controllers/config.go

```diff
@@ -102,6 +112,54 @@ func InitFromFlags(fs *flag.FlagSet, cfg *ManagerConfig) {
 
 	cfg.SecretControllerConfig = &secret.Config{}
 	secret.InitFromFlags(fs, cfg.SecretControllerConfig)
+
+	return nil
+}
+
+// InitFeatureGates initializes feature gates from the provided CLI flag set.
+func InitFeatureGates(fs *flag.FlagSet, featureFlags *featuregate.MutableFeatureGate) error {
+	*featureFlags = featuregate.NewFeatureGate()
```

Comment:
Why are you resetting the `featureGates` variable to a new feature gate after passing it as a parameter from the function call? Instead, you should simply pass manager config to this function, and use a separate variable to initialize the feature gates flag and then run `AddFlag` and then assign it to `cfg.FeatureGates`.